database/clickhouse/clickhouse_test.go

```diff
@@ -63,115 +73,133 @@ func isReady(ctx context.Context, c dktest.ContainerInfo) bool {
 	return true
 }
 
-func Test(t *testing.T) {
-	dktesting.ParallelTest(t, specs, func(t *testing.T, c dktest.ContainerInfo) {
-		ip, port, err := c.Port(defaultPort)
-		if err != nil {
-			t.Fatal(err)
-		}
+func TestCases(t *testing.T) {
+	for _, engine := range tableEngines {
+		t.Run("Test_"+engine, testSimple(engine))
+		t.Run("Migrate_"+engine, testMigrate(engine))
+		t.Run("Version_"+engine, testVersion(engine))
+		t.Run("Drop_"+engine, testDrop(engine))
 
-		addr := clickhouseConnectionString(ip, port)
-		p := &clickhouse.ClickHouse{}
-		d, err := p.Open(addr)
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			if err := d.Close(); err != nil {
-				t.Error(err)
+	}
+}
+
+func testSimple(engine string) func(t *testing.T) {
```

Comment:
Hmmm, I haven't seen this pattern before!

What's the advantage of returning a test method? If we go with this approach, I'd rename the method since it's not actually performing the test. e.g. `func simpleTestFunc(engine string) func(*testing.T)`

Personally, I would have done:
`t.Run("Test_"+engine, func(t *testing.T) { testSimple(t, engine) })`

Or use a partial function:
`func withEngine(f func (*testing.T, string), engine string) func(*testing.T) { return func(t *testing.T) {f(t, engine)}}`

Reply:
I don't think it actually makes much difference in this case. Having your suggested pattern  of `t.Run("Test_"+engine, func(t *testing.T) { testSimple(t, engine) })` is potentially more flexible if we could reutilize them elsewhere.

Changed to your suggestion.